teps/0052-tekton-results-automated-run-resource-cleanup.md

```diff
@@ -0,0 +1,213 @@
+    - [Watcher Outages](#watcher-outages)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Alternatives](#alternatives)
+  - [Decouple Cleanup from Watcher](#decouple-cleanup-from-watcher)
+
+<!-- /toc -->
+
+## Summary
+
+This proposal details a solution to provide automatic TaskRun/PipelineRun
+resource clean up in conjunction with Tekton Results to preserve long term
+history while minimizing the footprint of data the Pipeline controller needs to
+keep track of.
+
+## Motivation
+
+Today, most users view Tekton Run history via the Tekton Pipeline API (e.g.
+kubectl, tkn, dashboard, etc). However since all CRD objects are
+[stored in etcd](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/),
+this creates a problem for the Pipeline controller - as the build history grows
+over time, so does the corpus of data the controller needs to handle in its data
+plane. Over time this grows unbounded, significantly degrading the performance
+of the cluster as more Runs are completed.
+
+Most of this data is unnecessary for the Pipeline controller to hold on to -
+once a Run is complete it no longer needs to be stored in etcd since the
+Pipeline controller will take no further action on it. With Tekton Results, we
+now have a place to store long term results independent of the Pipeline
+controller, even if the data is removed from the etcd data plane.
+
+Many users have implemented custom cleanup mechanisms to address this
+([including our own dogfooding cluster](https://github.com/tektoncd/plumbing/blob/2c1808d75b38d6fcaae0a37f5a9dda50545af4c8/tekton/resources/cd/cleanup-template.yaml)).
+This proposal aims to define a supported mechanism for this behavior in
+conjunction with our existing Results storage.
+
+Relevant Issues:
+
+- [tektoncd/pipeline#2332](https://github.com/tektoncd/pipeline/issues/2332)
+- [tektoncd/pipeline#1486](https://github.com/tektoncd/pipeline/pull/1486)
+- [tektoncd/pipeline#1429](https://github.com/tektoncd/pipeline/pull/1429)
+- [tektoncd/experimental#479](https://github.com/tektoncd/experimental/issues/479)
+- [tektoncd/pipeline#2452](https://github.com/tektoncd/pipeline/issues/2452)
+
+### Goals
+
+- Define a mechanism for cleaning up completed Runs to minimize resources the
+  Pipeline controller is responsible for.
+- Preserve Run metadata so that users can continue to query it beyond the
+  lifetime of the Pipeline controller.
+
+### Non-Goals
+
+- Providing a programmatic mechanism for end users to dynamically control deletion behavior.
+
+### Use Cases (optional)
+
+This is a commonly requested feature due to the strain Run history this puts on
+etcd over time. See [Relevant Issues under Motivation](#motivation).
+
+## Requirements
+
+- Run metadata should be available even after deletion from the Pipeline
```

Comment:
confirming: when you say "metadata" i think you mean the complete spec and status?

Reply:
ah okay so the ENTIRE object - just (pedantically) checking that we're not leaving some of the content out